<?xml version='1.0' encoding='UTF-8'?>
<robot name="stretch_description">
  <link name="link_wrist_yaw_bottom">
    <inertial>
      <origin xyz="-0.012462 -0.032314 -0.021973" rpy="0.0 0.0 0.0" />
      <mass value="0.24284" />
      <inertia ixx="0.000194" ixy="3.4e-05" ixz="7e-06" iyy="0.000107" iyz="2e-05" izz="0.000229" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw_bottom.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw_bottom.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_wrist_pitch">
    <inertial>
      <origin xyz="-0.006752 -0.01567 0.021672" rpy="0.0 0.0 0.0" />
      <mass value="0.208129" />
      <inertia ixx="8.5e-05" ixy="1.6e-05" ixz="5e-06" iyy="8.4e-05" iyz="4e-06" izz="0.000105" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_pitch.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_pitch.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_wrist_roll">
    <inertial>
      <origin xyz="6e-05 -3.1e-05 0.006242" rpy="0.0 0.0 0.0" />
      <mass value="0.01528" />
      <inertia ixx="1e-06" ixy="0.0" ixz="0.0" iyy="9.5e-07" iyz="0.0" izz="1.39e-06" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_roll.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_roll.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_straight_gripper">
    <inertial>
      <origin xyz="0.002409 -0.011138 0.028861" rpy="0.0 0.0 0.0" />
      <mass value="0.147853" />
      <inertia ixx="9.5e-07" ixy="-4e-06" ixz="-1e-06" iyy="0.000119" iyz="6e-06" izz="5.9e-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_straight_gripper.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_straight_gripper.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_grasp_center">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 -0.0 0.0" />
      <mass value="1.0" />
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
    </inertial>
  </link>
  <link name="base_link">
    <inertial>
      <origin xyz="-0.087526 -0.001626 0.081009" rpy="0.0 0.0 0.0" />
      <mass value="17.384389" />
      <inertia ixx="0.160002" ixy="0.006758" ixz="0.004621" iyy="0.138068" iyz="0.002208"
        izz="0.228992" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/base_link.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/base_link.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_mast">
    <inertial>
      <origin xyz="0.000337 0.722201 0.002652" rpy="0.0 0.0 0.0" />
      <mass value="1.764017" />
      <inertia ixx="0.291286" ixy="0.004149" ixz="-0.000174" iyy="0.001226" iyz="0.000354"
        izz="0.291429" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_mast.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_mast.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_lift">
    <inertial>
      <origin xyz="-0.020369 0.03438 0.02236" rpy="0.0 0.0 0.0" />
      <mass value="1.987167" />
      <inertia ixx="0.006518" ixy="-0.001539" ixz="0.000371" iyy="0.004464" iyz="0.000392"
        izz="0.008092" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_lift.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_lift.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_arm_l4">
    <inertial>
      <origin xyz="-0.000168 -0.000254 -0.125235" rpy="0.0 0.0 0.0" />
      <mass value="0.235626" />
      <inertia ixx="0.001518" ixy="-1e-06" ixz="3e-06" iyy="0.001519" iyz="3e-06" izz="0.000176" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l4.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l4.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_arm_l3">
    <inertial>
      <origin xyz="-1e-06 -0.000146 -0.11908" rpy="0.0 0.0 0.0" />
      <mass value="0.146817" />
      <inertia ixx="0.001081" ixy="0.0" ixz="0.0" iyy="0.001081" iyz="-1e-06" izz="7.7e-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l3.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l3.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_arm_l2">
    <inertial>
      <origin xyz="0.0 -7.1e-05 -0.115635" rpy="0.0 0.0 0.0" />
      <mass value="0.130565" />
      <inertia ixx="0.000943" ixy="0.0" ixz="0.0" iyy="0.000943" iyz="0.0" izz="5.7e-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l2.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l2.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_arm_l1">
    <inertial>
      <origin xyz="-1e-06 -0.000121 -0.113457" rpy="0.0 0.0 0.0" />
      <mass value="0.113658" />
      <inertia ixx="0.0008" ixy="0.0" ixz="0.0" iyy="0.0008" iyz="0.0" izz="4e-05" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l1.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l1.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_arm_l0">
    <inertial>
      <origin xyz="0.033681 -0.000847 -0.031723" rpy="0.0 0.0 0.0" />
      <mass value="0.427734" />
      <inertia ixx="0.001689" ixy="-2.8e-05" ixz="0.000483" iyy="0.002107" iyz="-1.1e-05"
        izz="0.000571" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_arm_l0.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_arm_l0.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <link name="link_wrist_yaw">
    <inertial>
      <origin xyz="0.0 -3.9e-05 -0.016495" rpy="0.0 0.0 0.0" />
      <mass value="0.054422" />
      <inertia ixx="9e-06" ixy="0.0" ixz="0.0" iyy="9e-06" iyz="0.0" izz="7e-06" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/link_wrist_yaw.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link>
  <!-- <link name="base_imu">
    <inertial>
      <origin xyz="0.00300349280517617 0.00149777182047641 -0.00193103885249443" rpy="0.0 0.0 0.0" />
      <mass value="0.00092" />
      <inertia ixx="3e-08" ixy="0.0" ixz="0.0" iyy="5e-08" iyz="0.0" izz="8e-08" />
    </inertial>
    <visual>
      <geometry>
        <mesh filename="./meshes/base_imu.STL" />
      </geometry>
      <material name="" />
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </visual>
    <collision>
      <geometry>
        <mesh filename="./meshes/base_imu.STL" />
      </geometry>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
    </collision>
  </link> -->
  <link name="link_base_translation">
    <inertial>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 -0.0 0.0" />
      <mass value="1.0" />
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="1.0" iyz="0.0" izz="1.0" />
    </inertial>
  </link>
  <joint name="joint_wrist_yaw_bottom" type="fixed">
    <parent link="link_wrist_yaw" />
    <child link="link_wrist_yaw_bottom" />
    <axis xyz="nan nan nan" />
    <origin xyz="0.0 0.0 0.0" rpy="-3.14159265358979 1.13367999021379e-14 1.57079632679489" />
  </joint>
  <joint name="joint_wrist_pitch" type="revolute">
    <limit effort="100.0" velocity="1.0" lower="-1.57" upper="0.56" />
    <parent link="link_wrist_yaw_bottom" />
    <child link="link_wrist_pitch" />
    <axis xyz=" 0.  0. -1." />
    <origin xyz="0.0 -0.0195500000000002 -0.0247499999999984"
      rpy="1.5707963267949 -8.12895570882604e-15 -3.14159265358979" />
  </joint>
  <joint name="joint_wrist_roll" type="revolute">
    <limit effort="100.0" velocity="1.0" lower="-3.14" upper="3.14" />
    <parent link="link_wrist_pitch" />
    <child link="link_wrist_roll" />
    <axis xyz="0. 0. 1." />
    <origin xyz="-0.0188587444076125 -0.0239999999998942 0.01955"
      rpy="3.14159265358979 1.5707963267948966 0.0" />
  </joint>
  <joint name="joint_straight_gripper" type="fixed">
    <parent link="link_wrist_roll" />
    <child link="link_straight_gripper" />
    <axis xyz="nan nan nan" />
    <origin xyz="0.0 0.0 0.0155" rpy="3.54987407349455e-30 3.24021254484265e-20 -3.14159265358979" />
  </joint>
  <joint name="joint_grasp_center" type="fixed">
    <parent link="link_straight_gripper" />
    <child link="link_grasp_center" />
    <axis xyz="1. 0. 0." />
    <origin xyz="0.0 0.0 0.23" rpy="-1.5707963267949 -1.5707963267948966 0.0" />
  </joint>
  <joint name="joint_mast" type="fixed">
    <parent link="link_base_translation" />
    <child link="link_mast" />
    <axis xyz="nan nan nan" />
    <origin xyz="-0.0682070210503061 0.1271855564797244 0.0326851200442547"
      rpy="1.5501272264241974 -0.003518973808225745 -0.009269475816773206" />
  </joint>
  <joint name="joint_lift" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="1.1" />
    <parent link="link_mast" />
    <child link="link_lift" />
    <axis xyz="0. 0. 1." />
    <origin xyz="-0.037385 0.1666 0.0"
      rpy="1.5707926535897934 1.5707926535755432 -3.141592653589793" />
  </joint>
  <joint name="joint_arm_l4" type="fixed">
    <parent link="link_lift" />
    <child link="link_arm_l4" />
    <axis xyz="nan nan nan" />
    <origin xyz="-0.2560171715199499 -0.007652579073783216 -0.005285974837581016"
      rpy="1.59232809369906 -0.036574035521176995 -1.5500406624491974" />
  </joint>
  <joint name="joint_arm_l3" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13" />
    <parent link="link_arm_l4" />
    <child link="link_arm_l3" />
    <axis xyz="0. 0. 1." />
    <origin xyz="0.0 0.0 0.013" rpy="7.68831233799385e-30 2.36716479416092e-30 2.29652732251143e-17" />
  </joint>
  <joint name="joint_arm_l2" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13" />
    <parent link="link_arm_l3" />
    <child link="link_arm_l2" />
    <axis xyz="0. 0. 1." />
    <origin xyz="0.0 0.0 0.013" rpy="0.0 1.57655765344625e-30 -1.66533453693773e-16" />
  </joint>
  <joint name="joint_arm_l1" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13" />
    <parent link="link_arm_l2" />
    <child link="link_arm_l1" />
    <axis xyz="0. 0. 1." />
    <origin xyz="0.0 0.0 0.0129999999999981"
      rpy="-7.63746778746202e-30 -7.88860905221012e-31 1.11022302462516e-16" />
  </joint>
  <joint name="joint_arm_l0" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="0.0" upper="0.13" />
    <parent link="link_arm_l1" />
    <child link="link_arm_l0" />
    <axis xyz="0. 0. 1." />
    <origin xyz="0.0 0.0 -0.0137499999991968"
      rpy="7.63746778746202e-30 -3.80121128864402e-15 2.62707547767438e-15" />
  </joint>
  <joint name="joint_wrist_yaw" type="revolute">
    <limit effort="100.0" velocity="1.0" lower="-1.75" upper="4.0" />
    <parent link="link_arm_l0" />
    <child link="link_wrist_yaw" />
    <axis xyz=" 0.  0. -1." />
    <origin xyz="0.0830000000000654 -0.0307500000000129 0.0" rpy="1.5708 4.2595e-14 2.6415e-15" />
  </joint>
  <!-- <joint name="joint_base_imu" type="fixed">
    <parent link="base_link" />
    <child link="base_imu" />
    <axis xyz="nan nan nan" />
    <origin xyz="-0.12838 0.0031592 0.1474" rpy="3.1415853071795867 0.0 -1.5708" />
  </joint> -->
  <joint name="joint_base_translation" type="prismatic">
    <limit effort="100.0" velocity="1.0" lower="-1.0" upper="1.0" />
    <parent link="base_link" />
    <child link="link_base_translation" />
    <axis xyz="1. 0. 0." />
    <origin xyz="0.0 0.0 0.0" rpy="0.0 -0.0 0.0" />
  </joint>
  <material name="aluminum">
    <color rgba="0.5 0.5 0.5 1. " />
  </material>
  <material name="plastic">
    <color rgba="0.1 0.1 0.1 1. " />
  </material>
  <material name="">
    <color rgba="0.79215686 0.81960784 0.93333333 1.        " />
  </material>
</robot>